### ISSUE TITLE
Cannot open load file: No such file or directory, flycheck

### ISSUE THREAD

**Issue by timo-a** [emacs-lsp/lsp-ui#559]
I just installed `lsp-ui` (after installing only `lsp-mode` and `lsp-haskell`) and `*Compile-Log*` says:
```
^L
Compiling file /home/[...]/.emacs.d/elpa/lsp-ui-20210115.1200/lsp-ui-flycheck.el at Mon Jan 18 00:10:44 2021
lsp-ui-flycheck.el:27:1:Error: Cannot open load file: No such file or directory, flycheck
```

I have indeed not installed `flycheck` before, but as  a (inexperienced) user I would expect `flycheck` to be installed automatically if it is a dependency. (And if it isn't a dependency or optional, I would expect just a warning not an error)

**Comment by jcs090218** [comment]
This issue is duplicate to #431, #555, #558.

The current suggested solution, see #514.

**Comment by jcs090218** [comment]
I have applied fix in #560. 

Feel free to reopen this if the issue remains. Thanks!
